pkg/applicationserver/applicationserver.go

```diff
@@ -600,7 +807,6 @@ func (as *ApplicationServer) handleUplink(ctx context.Context, ids ttnpb.EndDevi
 				return nil, nil, errDeviceNotFound.WithAttributes("device_uid", unique.ID(ctx, ids))
 			}
 			if dev.Session == nil || !bytes.Equal(dev.Session.SessionKeyID, uplink.SessionKeyID) {
```

Comment:
The nesting here is very difficult to understand, consider refactoring (removing unnecessary nesting) and/or adding comments